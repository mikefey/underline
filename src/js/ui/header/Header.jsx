import React from 'react';
import style from './Header.scss';

/**
 * Header component
 */
const Header = (props) => {
  return (
    <div className={style.wrapper}>
      <div className={style.content}>
        {props.leftButton}
        <h2 className={style.title}>{props.text}</h2>
        {props.rightButton}
      </div>
    </div>
  );
};

Header.defaultProps = {
  leftButton: null,
  rightButton: null,
};


/**
 * PropTypes
 * @prop {ReactElement} leftButton - A button component to display on the left
 * side of the header
 * @prop {ReactElement} rightButton - A button component to display on the right
 * side of the header
 * @prop {String} text - The text to display in the header
 */
Header.propTypes = {
  leftButton: React.PropTypes.element,
  rightButton: React.PropTypes.element,
  text: React.PropTypes.string.isRequired,
};

export default Header;
