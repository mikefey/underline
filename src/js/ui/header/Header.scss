@import './../../../styles/colors';
@import './../../../styles/variables';

.wrapper {
  height: 60px;
  left: 0;
  position: fixed;
  text-align: center;
  width: 100%;
  z-index: 600;
}

.content {
  background-color: $white;
  border-bottom: 1px solid $dark-grey;
  display: inline-block;
  height: 100%;
  position: relative;
  max-width: $max-content-width;
  width: 100%;
}

.title {
  color: $dark-grey;
  font-size: 18px;
  margin: 16px 0 0;
}
