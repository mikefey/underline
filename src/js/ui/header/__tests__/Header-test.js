import test from 'ava';
import React, { Component } from 'react';
import { renderToString } from 'react-dom/server';

import Header from './../Header.jsx';

test('Header component', (t) => {
  const tree = renderToString(<Header text={'Your books'} />);
  t.snapshot(tree);
});
