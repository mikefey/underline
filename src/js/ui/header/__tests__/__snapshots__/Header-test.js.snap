exports[`Header component 1`] = `"\"<div class=\\\"Header__wrapper___2PTLf_nkW7S2HEU\\\" data-reactroot=\\\"\\\" data-reactid=\\\"1\\\" data-react-checksum=\\\"-46641228\\\"><div class=\\\"Header__content___2VIdu565Foup5Pb\\\" data-reactid=\\\"2\\\"><h2 class=\\\"Header__title___1_ebCiqkttbS44m\\\" data-reactid=\\\"3\\\">Your books</h2></div></div>\""`;
